osquery/core/init.cpp

```diff
@@ -249,42 +238,30 @@ static inline void printUsage(const std::string& binary, ToolType tool) {
   fprintf(stdout, EPILOG);
 }
 
-void Initializer::platformSetup() {
-// Initialize the COM libraries utilized by Windows WMI calls.
-#ifdef WIN32
-  auto ret = ::CoInitializeEx(0, COINIT_MULTITHREADED);
-  if (ret != S_OK) {
-    ::CoUninitialize();
-  }
-#else
-#endif
-}
-
-void Initializer::platformTeardown() {
-// Before we shutdown, we must insure to free the COM libs in windows
-#ifdef WIN32
-  ::CoUninitialize();
-#else
-#endif
-}
-
 Initializer::Initializer(int& argc, char**& argv, ToolType tool)
-    : argc_(&argc),
-      argv_(&argv),
-      tool_(tool),
-      binary_((tool == ToolType::DAEMON) ? "osqueryd" : "osqueryi") {
+    : argc_(&argc), argv_(&argv) {
+  // Initialize random number generated based on time.
   std::srand(static_cast<unsigned int>(
       chrono_clock::now().time_since_epoch().count()));
   // The config holds the initialization time for easy access.
   Config::setStartTime(getUnixTime());
 
-  // Initialize registries and plugins
-  registryAndPluginInit();
+  // osquery can function as the daemon or shell depending on argv[0].
+  if (tool == ToolType::SHELL_DAEMON) {
+    if (boost::filesystem::path(argv[0]).filename() == "osqueryd") {
```

Comment:
Maybe contains would be better? On Windows this'll be `osqueryd.exe`, and thus we'll always get shell behavior.